client/state/utils.js

```diff
@@ -19,3 +23,34 @@ export function isValidStateWithSchema( state, schema, checkForCycles = false, b
 	}
 	return result.valid;
 }
+
+export function createReducer( initialState = null, handlers = {}, schema = null ) {
+	const defaultHandlers = {
+		[SERIALIZE]: ( state ) => {
+			if ( schema !== null ) {
+				return state;
+			}
+
+			return initialState;
+		},
+		[DESERIALIZE]: ( state ) => {
+			if ( schema !== null && isValidStateWithSchema( state, schema ) ) {
+				return state;
+			}
+
+			return initialState;
+		}
+	};
+
+	handlers = Object.assign( {}, defaultHandlers, handlers );
```

Comment:
> Ah, I guess what your original suggestion enables is not serializing state unless a schema is defined?

Yes, that's why I added a schema check inside default handlers. I wanted to make sure that by default you don't persist your state as it is unless the schema is defined. This should help at least communicating that you need to provide the schema or consciously provide your own handling for `SERIALIZE` and `DESERIALIZE` actions.